### DISCUSSION TITLE
Two Components inside of One <Controller> wrapper

### DISCUSSION THREAD

**Discussion by Brian-1150** [react-hook-form/react-hook-form#6340] [Q&A]
For example, react-datepicker requires two DatePicker components in order to build a RangePicker.  Can I put both DatePickers in one Controller?

**Comment by bluebill1049** [comment]
yes, you can. `Controller` is nothing special but collect value and validate them.

```
const YourDateComponent = () => {
  const [x, y] = useState();

  return <div>
    <YourDatePickA x />
    <YourDatePickB x />
  </div>
}

<Controller render={() => <YourDateComponent />} />
```
